src/emulator/storage/files.ts

```diff
@@ -149,14 +240,58 @@ export class StorageLayer {
    * @throws {ForbiddenError} if request is unauthorized
    */
   public async getObject(request: GetObjectRequest): Promise<GetObjectResponse> {
+    let authorized = true;
     const metadata = this.getMetadata(request.bucketId, request.decodedObjectId);
 
     // If a valid download token is present, skip Firebase Rules auth. Mainly used by the js sdk.
     const hasValidDownloadToken = (metadata?.downloadTokens || []).includes(
       request.downloadToken ?? ""
     );
-    let authorized = hasValidDownloadToken;
-    if (!authorized) {
+
+    let checkAuth = false;
+
+    if (!hasValidDownloadToken) {
+      if (request.urlSignature) {
+        const prevSignature = request.urlSignature;
+
+        if (!request.urlUsableSeconds || !request.urlTtlSeconds) {
+          throw new BadRequestError(`Invalid ${!request.urlUsableSeconds ? "Date" : "TTL"}`);
+        }
+        const start = convertDateToMS(request.urlUsableSeconds);
+
+        if (!start || !Number.isInteger(request.urlTtlSeconds)) {
+          throw new BadRequestError(`Invalid ${!start ? "Date" : "TTL"}`);
+        }
+
+        const changeInTime = request.urlTtlSeconds * SECONDS_TO_MS_FACTOR;
+        const end = start + changeInTime;
+        const now = convertDateToMS(this.getCurrentDate());
+
+        const isLive = now >= start && now < end;
+
+        if (!isLive) {
+          throw new BadRequestError("Url has Expired");
+        }
+
+        const unsignedUrl = createUnsignedUrl({
+          bucketId: request.bucketId,
+          decodedObjectId: request.decodedObjectId,
+          url: request.url as string,
+          urlUsableSeconds: request.urlUsableSeconds,
+          urlTtlSeconds: request.urlTtlSeconds,
+        });
+
+        const isCorrect = createSignature(unsignedUrl) === prevSignature;
```

Comment:
You can inline this:
```
if (createSignature(unsignedUrl) !== prevSignature) {
   throw new ForbiddenError("Invalid signed url signature")
} else {
  ...
}
```

Reply:
for clarity purposes, i think having a variable would be easier to read. Especially since we don't inline the isLive variable a few lines above.